src/daft-dsl/src/lit.rs

```diff
@@ -68,6 +69,8 @@ pub enum LiteralValue {
     /// Python object.
     #[cfg(feature = "python")]
     Python(PyObjectWrapper),
+
+    Struct(HashMap<Field, LiteralValue>),
```

Comment:
Storing the fields in struct as a hashmap will change the ordering. You should use a vec or indexmap instead